docs/src/main/asciidoc/hibernate-search-elasticsearch.adoc

```diff
@@ -531,9 +531,9 @@ quarkus.hibernate-orm.sql-load-script=import.sql <4>
 
 quarkus.hibernate-search.elasticsearch.version=7 <5>
 quarkus.hibernate-search.elasticsearch.analysis.configurer=org.acme.hibernate.search.elasticsearch.config.AnalysisConfigurer <6>
-quarkus.hibernate-search.elasticsearch.automatic-indexing.synchronization.strategy=searchable <7>
-quarkus.hibernate-search.elasticsearch.index-defaults.lifecycle.strategy=drop-and-create <8>
-quarkus.hibernate-search.elasticsearch.index-defaults.lifecycle.required-status=yellow <9>
+quarkus.hibernate-search.elasticsearch.index-defaults.lifecycle.strategy=drop-and-create <7>
+quarkus.hibernate-search.elasticsearch.index-defaults.lifecycle.required-status=yellow <8>
+quarkus.hibernate-search.automatic-indexing.synchronization.strategy=searchable <9>
```

Comment:
TBH, I'm not totally sold on this change. It makes the configuration all weird.

I'm also surprised it's not something you can configure per backend in Search.

Reply:
Synchronization is about:

1. Whether to force a refresh on the backend side before considering indexing "done" (`committed` vs. `searchable`).
2. Whether we wait for indexing to be "done" after a transaction, or we just proceed without one works are queued (`queued` vs. `committed`/`searchable`).

Number 1 could theoretically be configured per backend, or even per index, though I don't really see the point.
Number 2 definitely cannot.

Maybe at some point we could split that into two separate configuration options, number 1 being configurable per backend while number 2 would not. However, that would introduce nonsensical combinations (do not wait for indexing to be "done", but force a refresh before considering it's "done"...). I also think it would make it even harder to explain the whole concept, but maybe that's just me.

Reply:
What bugs me a bit tbh is having this property outside of the `elasticsearch` context.

The other ones we had there were for advanced configurations. Here the default configuration looks a bit asymmetrical.